doc/managing_pachyderm/deploy_troubleshooting.md

```diff
@@ -0,0 +1,121 @@
+
+First get your cluster name. This will be in the deploy logs you saved from running `aws.sh` (if you utilized the [one shot deployment](http://docs.pachyderm.io/en/latest/deployment/amazon_web_services.html#one-shot-script)), or can be retrieved via `kops get clusters`.
+
+Then you'll need to grab the new public IP address of your master node. The master node will be named something like `master-us-west-2a.masters.somerandomstring.kubernetes.com`
+
+Update the etc hosts entry in `/etc/hosts` such that the api endpoint reflects the new IP, e.g:
+
+```
+54.178.87.68 api.somerandomstring.kubernetes.com
+```
+
+### One shot script never completes
+
+#### Symptom
+
+The `aws.sh` one shot deploy script hangs on the line:
+
+```
+Retrieving ec2 instance list to get k8s master domain name (may take a minute)
+```
+
+If it's been more than 10 minutes, there's likely an error.
+
+#### Recourse
+
+Check the AWS web console / autoscale group / activity history. You have probably hit an instance limit.  To navigate there, open the AWS web console for EC2. Check to see if you have any instances with names like::
+
+```
+master-us-west-2a.masters.tfgpu.kubernetes.com
+nodes.tfgpu.kubernetes.com
+```
+
+If not, navigate to "Auto Scaling Groups" in the left hand menu. Then find the ASG with your cluster name:
+
+```
+master-us-west-2a.masters.tfgpu.kubernetes.com
+```
+
+Look at the "Activity History" in the lower pane. More than likely, you'll see a "Failed" error message describing why it failed to provision the VM. You're probably run into an instance limit for your account for this region. If you're spinning up a GPU node, make sure that your region supports the instance type you're trying to spin up.
+
+A successful provisioning message looks like:
+
+```
+Successful
+Launching a new EC2 instance: i-03422f3d32658e90c
+2017 June 13 10:19:29 UTC-7
+2017 June 13 10:20:33 UTC-7
+Description:DescriptionLaunching a new EC2 instance: i-03422f3d32658e90c
+Cause:CauseAt 2017-06-13T17:19:15Z a user request created an AutoScalingGroup changing the desired capacity from 0 to 1. At 2017-06-13T17:19:28Z an instance was started in response to a difference between desired and actual capacity, increasing the capacity from 0 to 1.
+```
+
+While a failed one looks like:
+
+```
+Failed
+Launching a new EC2 instance
+2017 June 12 13:21:49 UTC-7
+2017 June 12 13:21:49 UTC-7
+Description:DescriptionLaunching a new EC2 instance. Status Reason: You have requested more instances (1) than your current instance limit of 0 allows for the specified instance type. Please visit http://aws.amazon.com/contact-us/ec2-request to request an adjustment to this limit. Launching EC2 instance failed.
+Cause:CauseAt 2017-06-12T20:21:47Z an instance was started in response to a difference between desired and actual capacity, increasing the capacity from 0 to 1.
+```
+### VPC Limit Exceeded
+
+#### Symptom
+
+When running `aws.sh` one shot script or otherwise deploying Pachyderm, you will see:
```

Comment:
You'll only see this if you're using kops. So I think you can remove the bit after the 'or' clause